apis/postgresql/v1alpha1/role_types.go

```diff
@@ -83,13 +83,28 @@ type RoleParameters struct {
 	// for this role. If no reference is given, a password will be auto-generated.
 	// +optional
 	PasswordSecretRef *xpv1.SecretKeySelector `json:"passwordSecretRef,omitempty"`
+
+	// ConfigurationParameters to be applied to the role. If specified, any other configuration parameters set on the
+	// role in the database will be reset.
+	//
+	// See https://www.postgresql.org/docs/current/runtime-config-client.html for some available configuration parameters.
+	// +optional
+	ConfigurationParameters []RoleConfigurationParameter `json:"configurationParameters,omitempty"`
```

Comment:
Nit: Kubernetes API guidelines require nil-able optional fields.

Perhaps a `RoleConfigurationParameterList` might be useful here:

```
type RoleConfigurationParameterList []RoleConfigurationParameter
```

Then we can do something like this:

```
ConfigurationParameters *RoleConfigurationParameterList `json:"configurationParameters,omitempty"`
```

Reply:
Cool, I wasn't aware of this. However I can't get `make generate` to work if I make that change. Not too familiar with this stuff :D 
`angryjet: error: error loading packages using pattern ./...: git/petteja/provider-sql/apis/postgresql/v1alpha1/zz_generated.deepcopy.go:766:10: cannot use new([]RoleConfigurationParameter) (value of type *[]RoleConfigurationParameter) as *RoleConfigurationParameterList value in assignment`

Any clues?

Reply:
Couldn't get it working with type aliasing, might be a bug in deepcopy generation. Changed to pointer type without aliasing.

Reply:
Gotcha, great that you got it working. Will review again later today

Reply:
The test seems to fail because of https://github.com/crossplane-contrib/provider-sql/pull/73